prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/channel/ChannelAddCommand.java

```diff
@@ -43,36 +45,55 @@ public class ChannelAddCommand extends AbstractCommand {
     @CommandLine.Option(names = CliConstants.CHANNEL_NAME, required = true)
     private String channelName;
 
-    @CommandLine.Option(names = CliConstants.CHANNEL_MANIFEST, required = true)
-    private String gavUrlOrPath;
+    @CommandLine.ArgGroup(exclusive = false)
+    ChannelGroup channelGroup;
+
+    static class ChannelGroup {
+        @CommandLine.Option(names = CliConstants.CHANNEL_MANIFEST)
+        private String manifestLocation;
```

Comment:
This can be a Path rather then a String

Reply:
I checking the windows failures and have a suspicion that this change to Path causes problems.

```
Invalid value for option '--manifest': cannot convert 'org.test:test' to interface java.nio.file.Path (java.nio.file.InvalidPathException: Illegal char <:> at index 8: org.test:test)
```
https://github.com/wildfly-extras/prospero/actions/runs/9114495539/job/25058577001?pr=668

I think the core issue is that manifestLocation should remain a String, because it may not be a path, it can still be a maven GAV..?

Reply:
ugh sorry, my bad that was meant to be a comment about `channelLocation` not `manifestLocation`

Reply:
And if that is true, and --manifest can be a GAV, then the same rules should probably apply to --channel (alias channelLocation).

Currently I can see channelLocation being still a String, but later in the code being treated as path:

```
ChannelAddCommand
...
public Integer call() throws Exception {
        Path installationDirectory = determineInstallationDirectory(directory);
        List<Channel> channels = null;
        if (channelOptions.channelLocation != null && !channelOptions.channelLocation.isEmpty()) {
            String yaml = Files.readString(Path.of(channelOptions.channelLocation));
            channels = ChannelMapper.fromString(yaml);
            Channel channel = channels.get(0);
            channels.set(0, new Channel(channelName, channel.getDescription(), channel.getVendor(), channel.getRepositories(), channel.getManifestCoordinate(), channel.getBlocklistCoordinate(), channel.getNoStreamStrategy()));
            if (channels.size() > 1) {
                throw new RuntimeException(CliMessages.MESSAGES.sizeOfChannel());
            }
        } else if ...
```

and this is I think caused by us giving wrong instructions to @parsharma. She originally used the resolver to obtain the channel instance, which IMO was correct, if channelLocation can be a GAV. Then we told her to use ChannelMapper which only works for local paths.

WDYT @spyrkob ?

Reply:
Yes, for simplicity I'd rather go with local channels only - at least initially. When resolving channels from GAV we'd need to add options like offline mode and repositories (which gets even more confusing with existing repositories parameter).

At very least I think this should be two separate parameters - maybe something like `--channel-file` and `--channel-gav` - it's much easier to explain it in documentation this way (and less buggy to handle in code).

Reply:
So for @parsharma , this means:

1. change manifestLocation back to String :),
2. change channelLocation to Path, and leave the rest as is.

Reply:
Considerations out-of-scope for this PR:

I agree the repositories parameter is bit overloaded. This may be similar even in the install command however. One thing is which repository the channel / manifest should be looked up, another where the channel artifacts should be resolved from.

The split to --channel-file, --channel-gav can make sense, but if it's ever done if should probably be done everywhere. Which means big change for users. Having just single command take different kind of parameters from all the other commands is weird.

Maybe refactoring the resolution logic for the current universal gav/url/path parameters, so they are easier to reuse.